### PR TITLE
Seeding Specs: Only mock ENV vars needed 

### PR DESCRIPTION
**Pull request by weedySeaDragon**
### PT Story: only stub ENV vars needed for tests
https://www.pivotaltracker.com/story/show/161967924

### Changes proposed in this pull request:
1.  Changed how ENV is stubbed so that only the entries being tested are stubbed, not the entire ENV
2. seed_development_spec.rb and seed_production_spec.rb had a huge amount of duplicated code between them.  I refactored out the shared code.
3. I created ENV_... constants not just because _constants are good_ but also to make it easier to spot them when we later clean up all of the ENV vars
4. I renamed the top specs so that they're more descriptive

Ready for review:
@patmbolger @thesuss

### CODE REVIEW

File: spec/db/shared_specs_db_seeding.rb [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    it 'admin is in the db' do
+      expect(admin_in_db.admin).to be_truthy
+    end
+
+    it "admin email is = #{admin_email}" do
+      expect(admin_in_db.email).to eq(admin_email)
+    end
+
+    it "admin email is = ENV['SHF_ADMIN_PWD']" do
```
Comment (patmbolger) [inline spec/db/shared_specs_db_seeding.rb]: perhaps 'admin password is in the db' ?

File: spec/db/shared_specs_db_seeding.rb [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    it 'admin email not found' do
+      admin_email_value = ENV.delete(ENV_ADMIN_EMAIL_KEY)
+      expect { Rails.application.load_seed }.to raise_exception SeedHelper::SeedAdminENVError
+      ENV[ENV_ADMIN_EMAIL_KEY] = admin_email_value
+    end
+
+    it 'admin email is an empty string' do
+      stub_const('ENV', ENV.to_hash.merge({ENV_ADMIN_EMAIL_KEY => ''}) )
+      expect { Rails.application.load_seed }.to raise_exception SeedHelper::SeedAdminENVError
```
Comment (patmbolger) [inline spec/db/shared_specs_db_seeding.rb:86]: Do you need to reset the email ENV at the end of this test?  (same for similar test below)

**Review by patmbolger — COMMENTED** [review]
Looks good - please see comments.

File: spec/db/shared_specs_db_seeding.rb (hunk 2 above)
Comment (weedySeaDragon) [reply spec/db/shared_specs_db_seeding.rb:86]: Nope.  RSpec mocks takes care of that:  

> All rspec-mocks constructs have a per-example lifecycle. Message expectations are verified
after each example. Doubles, method stubs, stubbed constants, etc. are all cleaned up after
each example. This ensures that each example can be run in isolation, and in any order.

(from https://relishapp.com/rspec/rspec-mocks/v/3-8/docs/basics/scope)

**Review by patmbolger — APPROVED** [review]

**Pull request merged**